uaclient/api/u/pro/attach/magic/wait/v1.py

```diff
@@ -69,8 +70,17 @@ def wait(
             )
         except exceptions.MagicAttachTokenError:
             break
-
-        if wait_resp.get("contractToken") is not None:
+        # If we have a flaky connectivity error, this part of the code
+        # will make sure that we try at least three more times before
+        # raising a ConnectivityError.
+        except exceptions.ConnectivityError as e:
+            if num_connection_errors < 3:
+                num_connection_errors += 1
+                wait_resp = None
```

Comment:
I guess you could also use `continue` here, but it's up to you of course.

Reply:
My only concern with `continue` here is that we would not call sleep and increment the number of attempts. That's why I am using that approach

Reply:
Oh you're right. I read the code wrong. My bad!